sdk/python/kfp/_client.py

```diff
@@ -296,22 +297,23 @@ def list_pipelines(self, page_token='', page_size=10, sort_by=''):
     return self._pipelines_api.list_pipelines(page_token=page_token, page_size=page_size, sort_by=sort_by)
 
   # TODO: provide default namespace, similar to kubectl default namespaces.
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, namespace=None):
-    """Run a specified pipeline.
-
+  def _create_job_config(self, experiment_id, params, pipeline_package_path, pipeline_id, namespace):
+    """Create a JobConfig with spec and resource_references.
     Args:
       experiment_id: The string id of an experiment.
-      job_name: name of the job.
-      pipeline_package_path: local path of the pipeline package(the filename should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).
-      params: a dictionary with key (string) as param name and value (string) as as param value.
-      pipeline_id: the string ID of a pipeline.
-      namespace: kubernetes namespace where the pipeline runs are created.
+      pipeline_package_path: Local path of the pipeline package(the filename should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).
+      params: A dictionary with key (string) as param name and value (string) as param value.
+      pipeline_id: The string ID of a pipeline.
+      namespace: Kubernetes namespace where the pipeline runs are created.
         For single user deployment, leave it as None;
         For multi user, input a namespace where the user is authorized
-
     Returns:
-      A run object. Most important field is id.
+      A JobConfig object with attributes spec and resource_reference.
     """
+    class JobConfig:
```

Comment:
I'd define this outside of the function i think? since it's used outside the function, it can be called `_JobConfig` to keep it "private"

Reply:
hm the class `RunPipelineResult` is also defined inside the function `create_run_from_pipeline_package`.

Outside of the `_create_job_config` function the `JobConfig` is only used by calling their attributes, the `JobConfig` object is not created anywhere else though. I would say it's ok to define it inside the function as long as only instances of the class are used outside.

But I don't know what's best practice and I can't see any good practice in this code :laughing:

Reply:
ah indeed, no then it should be a similar style